graylog2-server/src/main/java/org/graylog2/inputs/transports/HttpTransport.java

```diff
@@ -118,6 +131,26 @@ private static Executor executorService(final String executorName, final String
         final LinkedHashMap<String, Callable<? extends ChannelHandler>> baseChannelHandlers =
                 super.getBaseChannelHandlers(input);
 
+        if (idleWriterTimeout > 0) {
+            // Install IdleState and IdleStateEvent handler to close idle connections after a timeout.
+            // This avoids dangling HTTP connections when the HTTP client does not close the connection properly.
+            // For details see: https://github.com/Graylog2/graylog2-server/issues/3223#issuecomment-270350500
+
+            baseChannelHandlers.put("idlestate", new Callable<ChannelHandler>() {
+                @Override
+                public ChannelHandler call() throws Exception {
+                    return new IdleStateHandler(timer, 0, idleWriterTimeout, 0, TimeUnit.SECONDS);
```

Comment:
Because I wasn't aware of it. Changed the code to use a ReadTimeoutHandler because I want to close the connection when the **client** does not write anymore. Got it mixed up. :wink: